query-engine/connector-test-kit/src/test/scala/writes/nestedMutations/notUsingSchemaBase/NestedConnectOrCreateMutationSpec.scala

```diff
@@ -437,381 +437,6 @@ class NestedConnectOrCreateMutationSpec extends FlatSpec with Matchers with ApiS
-         |            b_u: "B2"
-         |          }
-         |        }
-         |      }
-         |  }) {
-         |    id
-         |    oneB {
-         |      id
-         |    }
-         |  }
-         |}
-      """,
-      project,
-      2014,
-      errorContains = "The change you are trying to make would violate the required relation 'ModelAToModelB' between the `ModelA` and `ModelB` models.",
-      legacy = false,
-    )
-
-    // Child inlined cases
-    result = server.query(
-      s"""mutation {
-         |  createOneModelB(data: {
-         |    id: "B_id_2"
-         |    b_u: "B2"
-         |    oneA: {
-         |      connectOrCreate: {
-         |        where: { id: "A2" }
-         |        create: {
-         |          id: "A2",
-         |        }
-         |      }
-         |    }
-         |  }) {
-         |    id
-         |    oneA {
-         |      id
-         |    }
-         |  }
-         |}
-      """,
-      project,
-      legacy = false,
-    )
-
-    result.toString() should be("{\"data\":{\"createOneModelB\":{\"id\":\"B_id_2\",\"oneA\":{\"id\":\"A2\"}}}}")
-
-    // Updating existing parent (B), connecting existing child (A). Fails because the existing child would dangle, but has a required parent.
-    server.queryThatMustFail(
-      s"""mutation {
-         |  updateOneModelB(
-         |    where: { b_u: "B2" }
-         |    data: {
-         |      oneA: {
-         |        connectOrCreate: {
-         |          where: { id: "A2" }
-         |          create: {
-         |            id: "A2",
-         |          }
-         |        }
-         |      }
-         |  }) {
-         |    id
-         |    oneA {
-         |      id
-         |    }
-         |  }
-         |}
-      """,
-      project,
-      2014,
-      errorContains = "The change you are trying to make would violate the required relation 'ModelAToModelB' between the `ModelA` and `ModelB` models.",
-      legacy = false,
-    )
-
-    // New parent, connecting existing child. Same as above, fails because the existing child would dangle, but has a required parent.
-    server.queryThatMustFail(
-      s"""mutation {
-         |  createOneModelB(
-         |    data: {
-         |      id: "B_id_3"
-         |      b_u: "B3"
-         |      oneA: {
-         |        connectOrCreate: {
-         |          where: { id: "A2" }
-         |          create: {
-         |            id: "A2"
-         |          }
-         |        }
-         |      }
-         |  }) {
-         |    id
-         |    oneA {
-         |      id
-         |    }
-         |  }
-         |}
-      """,
-      project,
-      2014,
-      errorContains = "The change you are trying to make would violate the required relation 'ModelAToModelB' between the `ModelA` and `ModelB` models.",
-      legacy = false,
-    )
-  }
-
-  "A 1:1! relation connectOrCreate" should "work and prevent relation violations" taggedAs (IgnoreMsSql) in {
```

Comment:
This looks like a dedicated test case for required 1:1 relations that we don't need anymore.

Reply:
Jep, that can go.